src/main/java/com/singh/astha/notification/service/controllers/NotificationController.java

```diff
@@ -0,0 +1,44 @@
+package com.singh.astha.notification.service.controllers;
+
+import com.singh.astha.notification.service.dtos.common.NotificationRequest;
+import com.singh.astha.notification.service.dtos.response.wrapper.ResponseWrapper;
+import com.singh.astha.notification.service.services.NotificationService;
+import com.singh.astha.notification.service.utils.Constants;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import jakarta.validation.Valid;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/internal/notifications")
+@SecurityRequirement(name = "bearerAuth")
+public class NotificationController {
+
+    private final NotificationService notificationService;
+
+    public NotificationController(
+            NotificationService notificationService
+    ) {
+        this.notificationService = notificationService;
+
+    }
+
+    @PostMapping
+    public ResponseEntity<ResponseWrapper<String>> sendNotification(
+            @Valid @RequestBody NotificationRequest notificationRequest
+    ) {
+        notificationService.sendNotification(notificationRequest);
+        return ResponseEntity.ok().body(ResponseWrapper.success(Constants.SUCCESSFULLY_SENT_NOTIFICATION));
+    }
+
+    @PostMapping("/enqueue")
+    public ResponseEntity<ResponseWrapper<String>> enqueueToKafka(
+            @Valid @RequestBody NotificationRequest notificationRequest
+    ) {
+        notificationService.enqueueToKafka(notificationRequest);
+        return ResponseEntity.ok().body(ResponseWrapper.success(Constants.SUCCESSFULLY_ENQUEUED_MESSAGE_IN_KAFKA));
```

Comment:
Why let client know that we have enqueued the message in Kafka?

Just say `ENQUEUED_MESSAGE_IN_QUEUE`